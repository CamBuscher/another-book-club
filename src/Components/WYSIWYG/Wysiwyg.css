.wysiwyg {
  width: 75%;
  margin: auto;
  background-color: #ffffff80;
  padding: 20px;
}

.wysiwyg button { 
  margin-top: 10px;
}