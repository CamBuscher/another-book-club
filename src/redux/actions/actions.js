export const setUser = user => ({
  type: 'SET_USER',
  user
});

export const updateBookClubs = (club, clubName) => ({
  type: 'UPDATE_BOOK_CLUBS',
  club,
  clubName
});

export const removeBookClub = (clubName) => ({
  type: 'REMOVE_CLUB',
  clubName
});

export const setCurrentClub = club => ({
  type: 'SET_CURRENT_CLUB',
  club
});

export const addBookToClub = club => ({
  type: 'ADD_BOOK_TO_CLUB',
  club
});