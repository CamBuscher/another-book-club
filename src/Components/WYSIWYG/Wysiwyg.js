import React from 'react';
import ReactQuill from 'react-quill';
import 'react-quill/dist/quill.snow.css';
import './Wysiwyg.css';

class Wysiwyg extends React.Component {
  constructor(props) {
    super(props)
    this.state = { text: '' }
    this.handleChange = this.handleChange.bind(this)
  }

  handleChange(value) {
    this.setState({ text: value })
  }

  handleComment = () => {
    
  }

  render() {
    return (
      <div className='wysiwyg'>
        <h5>Leave a comment!</h5>
        <ReactQuill value={this.state.text}
          onChange={this.handleChange} />
        <button 
          onClick={this.handleComment}
        > Add Comment </button>
      </div>
    );
  }
}

export default Wysiwyg;