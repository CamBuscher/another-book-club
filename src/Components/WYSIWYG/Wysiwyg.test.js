import React from 'react';
import { shallow } from 'enzyme';
import { Wysiwyg, mapStateToProps, mapDispatchToProps } from './Wysiwyg';
import { addCommentToClub } from '../../redux/actions/actions';

describe('wysiwyg', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Wysiwyg addComment={jest.fn()} currentClub={{comments: []}}/>);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('calls dispatch with a addCommentToClub action', () => {
    const dispatch = jest.fn();

    const actionToDispatch = addCommentToClub();

    const mappedProps = mapDispatchToProps(dispatch);

    mappedProps.addComment();

    expect(dispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  it('should return the currentClub', () => {
    const mockState = {
      user: {
        bookClubs: [],
        id: 'asdasd'
      },
      currentClub: {}
    };

    let expected = { bookClubs: [], id: 'asdasd' };

    const mappedProps = mapStateToProps(mockState).user;

    expect(mappedProps).toEqual(expected);
  });
});