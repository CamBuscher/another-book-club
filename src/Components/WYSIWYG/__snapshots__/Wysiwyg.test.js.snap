// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wysiwyg should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Wysiwyg
    addComment={[Function]}
    currentClub={
        Object {
            "comments": Array [],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Quill
          modules={Object {}}
          onChange={[Function]}
          theme="snow"
          value=""
/>,
        <button
          onClick={[Function]}
>
           Add Comment 
</button>,
      ],
      "className": "wysiwyg",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "modules": Object {},
          "onChange": [Function],
          "theme": "snow",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Add Comment ",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": " Add Comment ",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Quill
            modules={Object {}}
            onChange={[Function]}
            theme="snow"
            value=""
/>,
          <button
            onClick={[Function]}
>
             Add Comment 
</button>,
        ],
        "className": "wysiwyg",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "modules": Object {},
            "onChange": [Function],
            "theme": "snow",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Add Comment ",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": " Add Comment ",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
